src/tasks/taskfilter.ts

```diff
@@ -1,42 +1,39 @@
 import { Task } from "./task";
-import { CompletedTask, FailedTask, SuccessfulSingleTask, SuccessfulTask, UncompletedTask } from "./taskinterfaces";
+import { CompletedTask, FailedTask, SuccessfulTask, TaskBase, UncompletedTask } from "./taskinterfaces";
 import { TaskSet } from "./taskset";
 
 export class TaskFilter {
-    public static isTask(value: unknown): value is (Task | TaskSet) {
-        if (!value) {
-            return false;
-        }
-        const task = value as (Task | TaskSet);
-        return task.isCompleted !== undefined && task.isFaulted !== undefined;
+    public static CompareFinishedTime(taskA: CompletedTask, taskB: CompletedTask) {
+        if (taskA.completionIndex > taskB.completionIndex) { return 1; }
+        if (taskA.completionIndex < taskB.completionIndex) { return -1; }
+        return 0;
     }
 
-    public static isSingleTask(task: Task | TaskSet): task is Task {
+    public static isYieldable(task: any): task is TaskBase {
+        return (task instanceof Task) || (task instanceof TaskSet);
```

Comment:
Unfortunately typescript interfaces are compile time only. Maybe a more robust method would be to check for the existence of all of the current properties on `TaskBase`. That way it would also apply to any new implementations of `TaskBase`. The downside is that if I removed a property from `TaskBase`, it would break the method.